WatchApp Extension/Scenes/GlucoseChartScene.swift

```diff
@@ -61,10 +61,11 @@ struct Scaler {
         return CGPoint(x: CGFloat(x.timeIntervalSince(startDate)) * xScale, y: CGFloat(y - glucoseMin) * yScale)
     }
 
-    func rect(for range: WatchDatedRange) -> CGRect {
+    func rect(for range: WatchDatedRange, minHeight: CGFloat = 2) -> CGRect {
         let a = point(range.startDate, range.minValue)
         let b = point(range.endDate, range.maxValue)
-        let size = CGSize(width: b.x - a.x, height: b.y - a.y)
+        // Enforce a minimum height so that narrow target ranges still show up:
```

Comment:
This comment doesn't make sense here, since Scaler is a implementing an API about rectangles. The fact that there's a minHeight implies that the user can specify a minimum height for whatever reason they want.

Two options:
1) Associate the comment with the default value
2) Put the comment in the place where it's relevant, which is where rect() is called

The fact that we have a default value is really just a shortcut that's breaking the API anyway. If you really want this to be clean, then the minHeight should have no default value and we should specify "2" in the caller AND comment the caller.

Then the caller is saying "make a rectangle between these two points, but since I know that I'm drawing target ranges- also make sure that the height is 2." That's worthy of a comment, and then the Scaler API is clean.

Reply:
I'd agree with you if this were a generic rectangle-drawing function, but it's not - despite the generic title, it's a function that is always drawing a WatchDatedRange.  So it doesn't make sense to imagine that there could be some other possible call to it, that would be drawing something else.  That's why I didn't have it in the call when I wrote it originally.  I still think it's cleaner that way, with just a minimum height in the CGSize call and a comment to explain it.

Reply:
Just to clarify a bit more, you're saying that the comment is relevant where minHeight is defined in the call, but I disagree - it becomes relevant when you get to the point where it's actually used.

Reply:
Sorry for the confusion and the excessive back and forth over what probably appears to be a very minor detail. I don't think it's minor though - which I'll discuss at the end of this response.

At a high level what I'm saying that putting a comment here is a bad code smell: the code should be able to stand on its own without this level of micro comment. In my prior comment, I was suggesting two different ways that we could address this issue. I think we have 3 options, which I'll walk through.

1) If you were to provide a minHeight variable, and you were to set it to zero by default (ie, not overly specializing for WatchDatedRange), then you were to override the default value in the caller (eg. `scaler.rect(for: range, minHeight: 2)`), you would put the comment where you do the override which would be in the caller. The comment would explain that if the minimum height is too small, the range is not visible and you'd probably only need one meta-comment for both calls.

2) If you want to keep the minHeight default value of 2 as part of the API then you would make the comment as part of the rect API call, eg:
```
// By default enforce a minimum height so that the range is visible
func rect(for range: WatchDatedRange, minHeight: CGFloat = 2) -> CGRect {
```

3) Revert the minHeight variable, embed 2 in the code and make it part of the rect API docs. You make a really good point that this is not a generic rect drawing function - it's specialized for WatchDatedRange, but the hardcoded constant should be part of the API definition, not an implementation detail. So if you go that route, it'd be:

```
// By default enforce a minimum height so that the range is visible
func rect(for range: WatchDatedRange) -> CGRect {
  ...
  let size = CGSize(width: b.x - a.x, height: (b.y - a.y, 2))
```

Any of these options resolves the problem for me, which is that we should always prefer contractual API level comments over implementation detail comments. If we don't hold the line on this, the code rapidly slides into excessive micro comments that take up space and harm maintainability.